sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/variant/VariantEvaluator.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.expressions.variant
+
+import scala.util.control.NonFatal
+
+import org.apache.spark.sql.catalyst.util.BadRecordException
+import org.apache.spark.sql.errors.QueryExecutionErrors
+import org.apache.spark.types.variant.{VariantBuilder, VariantSizeLimitException, VariantUtil}
+import org.apache.spark.unsafe.types.{UTF8String, VariantVal}
+
+object VariantEvaluator {
```

Comment:
Can we write it in Java?

Reply:
and `VariantExpressionEvalUtils` sounds a better name, if we keep using to implement variant related expressions.

Reply:
Let me try.

Reply:
> Can we write it in Java?

- If we write it in `Java`, we may have to write it like this:
<img width="751" alt="image" src="https://github.com/apache/spark/assets/15246973/077fdf50-d07b-4a60-b20a-e7cd7078a5ae">

- Otherwise, it's like this:
  <img width="760" alt="image" src="https://github.com/apache/spark/assets/15246973/4d76a8ac-bebe-491e-9997-17ab49922cc6">

- This will result in compile errors, as shown below:
  <img width="1000" alt="image" src="https://github.com/apache/spark/assets/15246973/caa68fe5-1950-4f46-8f3f-d1717580622e">
  <img width="970" alt="image" src="https://github.com/apache/spark/assets/15246973/1568f8a2-67b0-4a8a-8003-74c6b7704066">
  Because the class `StaticInvoke` did not consider that the method may `throw an exception` when codegen.
https://github.com/apache/spark/blob/87449c3f1d65a430fec60981e364e552165fa075/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/objects/objects.scala#L322-L340
(PS: Maybe we can use a separate PR to improve `StaticInvoke` later?)